src/runtime/rpc/asio_rpc_session.cpp

```diff
@@ -31,7 +31,6 @@ namespace tools {
 
 void asio_rpc_session::set_options()
 {
-    utils::auto_write_lock socket_guard(_socket_lock);
```

Comment:
delete is ok?

Reply:
Yes, there won't be race conditions after the change.